### PR TITLE
[BUG FIX] [MER-3708] 500 error with iframed lesson launch links

### PR DESCRIPTION
**Pull request by nicocirio**
[Link](https://eliterate.atlassian.net/browse/MER-3708) to the ticket

The query params in the URL were causing the iframe to crash. The fix removes the query params from adaptive pages routes and, since the `request_path` query param was needed to provide the back URL path, we now pass it through the session.

### Before
The request_path (used for the back/home button) was passed as an URL query param

https://github.com/user-attachments/assets/e1afcbf5-75ce-44b7-a9d2-ea0204e57d58

### After
The request_path is set to the conn session to keep back/home button functionality working as before

https://github.com/user-attachments/assets/c158de38-1320-4f67-ae83-ccb3c441edf9

### CODE REVIEW

File: lib/oli_web/plugs/redirect_by_attempt_state.ex [hunk 1]
```diff
@@ -222,22 +222,46 @@ defmodule OliWeb.Plugs.RedirectByAttemptState do
     end
   end
 
-  defp ensure_path(conn, rendering_type) when rendering_type in [:adaptive_lesson, :lesson] do
+  defp ensure_path(conn, rendering_type) when rendering_type == :lesson do
```
Comment (simonchoxx) [inline lib/oli_web/plugs/redirect_by_attempt_state.ex:225]: Here we could use pattern matching in the definition of the function, WDYT? 
I think it should work anyway by doing something like : 

`defp ensure_path(conn, :lesson) do`

File: lib/oli_web/plugs/redirect_by_attempt_state.ex [hunk 2]
```diff
@@ -222,22 +222,46 @@ defmodule OliWeb.Plugs.RedirectByAttemptState do
...
+      conn
+      |> halt()
+      |> assign(:already_been_redirected?, true)
+      |> Phoenix.Controller.redirect(
+        to: "/sections/#{section_slug}/lesson/#{revision_slug}?#{conn.query_string}"
+      )
+    end
+  end
+
+  defp ensure_path(conn, rendering_type) when rendering_type == :adaptive_lesson do
```
Comment (simonchoxx) [inline lib/oli_web/plugs/redirect_by_attempt_state.ex:245]: Same as above

**Review by simonchoxx — APPROVED** [review]
Nice solution by using the session to pass the `request_path` for the adaptive pages! 👏

I left some minor comments but they are not a priority to change.

File: lib/oli_web/plugs/redirect_by_attempt_state.ex (hunk 1 above)
Comment (nicocirio) [reply lib/oli_web/plugs/redirect_by_attempt_state.ex:225]: Totally agree! I adapted the code and didn't realize the overhead the `when` clause is adding here

**Pull request merged**